File: hw/tipantiza/u3/hw27ObserverPattern/ObserverPattern/src/main/java/ec/edu/espe/observerpattern/controller/StockController.java

package ec.edu.espe.observerpattern.controller;
import ec.edu.espe.observerpattern.model.IBM;
import ec.edu.espe.observerpattern.model.Investor;

/**
 *
 * @author Alexander Tipantiza, The Softwarrios, @ESPE
 */

public class StockController {
    public void runSimulation() {
        Investor s = new Investor("Soros");
        Investor b = new Investor("Berkshire");

        IBM ibm = new IBM("IBM", 120.00);
        ibm.addObserver(s);
        ibm.addObserver(b);

        ibm.setPrice(120.10);
        ibm.setPrice(121.00);
        ibm.setPrice(120.50);
        ibm.setPrice(120.75);
        ibm.setSymbol("IBMTEST");
    }
}

File: hw/tipantiza/u3/hw27ObserverPattern/ObserverPattern_py/controller/stock_controller.py
from model.ibm import IBM
from model.investor import Investor


class StockController:

    def run_simulation(self):

        soros = Investor("Soros")
        berkshire = Investor("Berkshire")

        ibm = IBM("IBM", 120.00)

        ibm.add_observer(soros)
        ibm.add_observer(berkshire) 

        ibm.set_price(120.10)
        ibm.set_price(121.00)
        ibm.set_price(120.50)
        ibm.set_price(120.75)
        ibm.set_symbol("IBMTEST")